test/libsolidity/semanticTests/operators/userDefined/literal_suffix_as_operator_definition.sol

```diff
@@ -0,0 +1,24 @@
+type B is bool;
+using {bitnot as ~, bitor as |} for B global;
+
+// NOTE: There are no literals of type B so these cannot be actually used on anything
+function bitnot(B x) pure suffix returns (B) {
+    return B.wrap(!B.unwrap(x));
+}
+
+function bitor(B x, B y) pure suffix returns (B) {
```

Comment:
Now that we have `suffix` specifier, we can check the parameters of the function and be more restrictive. 
I implemented that in #14015 and now this function signature generates a `TypeError` because the parameters must be of type `integer` and `unsigned integer`.

Reply:
Good point. The unary one will still work, but I wanted to actually disallow any types that are not implicitly convertible from literals so it will soon stop working anyway.

We should convert this to a syntax test and update expectations.